src/modules/fancyzones/lib/JsonHelpers.cpp

```diff
@@ -321,6 +327,37 @@ namespace JSONHelpers
         }
     }
 
+    void FancyZonesData::UpdatePrimaryDesktopData(const std::wstring& desktopId)
+    {
+        std::scoped_lock lock{ dataLock };
+        for (auto& [path, data] : appZoneHistoryMap)
+        {
+            if (ExtractVirtualDesktopId(data.deviceId) == DEFAULT_GUID)
+            {
+                data.deviceId = ReplaceVirtualDesktopId(data.deviceId, desktopId);
```

Comment:
I would prefer local lambda here instead of `ReplaceVirtualDecktopId` method

Reply:
Done.